### PR TITLE
Add `mypy` hook to pre-commit

### PR DESCRIPTION
**Pull request by slsh1o**
I'm not sure what else should I add to `mypy.ini`. Something related to tests or maybe to sphinx.

And also there are some errors which idk how to fix better.

Ref: #16

### CODE REVIEW

File: src/sphinxcontrib/towncrier/_scm_version.pyi [hunk 1]
```diff
@@ -0,0 +1 @@
+version: str
```
Comment (webknjaz) [inline src/sphinxcontrib/towncrier/_scm_version.pyi:1]: I'd also add a note that this is necessary because the actual module is autogenerated on build and may be missing at the time `mypy` runs.

File: mypy.ini [hunk 2]
```diff
@@ -0,0 +1,18 @@
+[mypy]
+python_version = 3.6
+color_output = True
+error_summary = True
+
+check_untyped_defs = True
+
+disallow_untyped_calls = False
```
Comment (webknjaz) [inline mypy.ini:8]: Does flipping this to True make it emit many errors? If not, do this and address the errors; if yes, submit it as a separate PR.

File: src/sphinxcontrib/towncrier/__init__.py [hunk 3]
```diff
@@ -5,9 +5,10 @@
 import sys
 from functools import lru_cache
 from pathlib import Path
-from typing import Any, Dict, List, Union
+from typing import Dict, List, Tuple, Union
 
 from sphinx.application import Sphinx
+from sphinx.config import Config
```
Comment (webknjaz) [inline src/sphinxcontrib/towncrier/__init__.py:11]: Let's do
```suggestion
from sphinx.config import Config as SphinxConfig
```
so that the code below is more readable.

File: mypy.ini (hunk 2 above)
Comment (slsh1o) [reply mypy.ini:8]: Turn it to True and there are no errors.

File: src/sphinxcontrib/towncrier/__init__.py [hunk 4]
```diff
@@ -66,7 +67,7 @@ def _get_changelog_draft_entries(
 
 
 @lru_cache(maxsize=1, typed=True)
-def _get_draft_version_fallback(strategy: str, sphinx_config: Dict[str, Any]):
+def _get_draft_version_fallback(strategy: str, sphinx_config: Config):
```
Comment (webknjaz) [inline src/sphinxcontrib/towncrier/__init__.py:70]: Also annotate the return type:
```suggestion
def _get_draft_version_fallback(strategy: str, sphinx_config: SphinxConfig) -> str:
```

File: mypy.ini (hunk 2 above)
Comment (webknjaz) [reply mypy.ini:8]: So keep it true. The goal is to make mypy setup as strict as possible.

Comment (webknjaz) [reply mypy.ini:8]: OTOH you could use https://mypy.readthedocs.io/en/stable/config_file.html to make a follow-up PR.

Comment (slsh1o) [reply mypy.ini:8]: Should I add some other configs?

Comment (webknjaz) [reply mypy.ini:8]: Yes, but probably in another PR.

File: src/sphinxcontrib/towncrier/__init__.py [hunk 5]
```diff
@@ -66,7 +67,7 @@ def _get_changelog_draft_entries(
 
 
 @lru_cache(maxsize=1, typed=True)
-def _get_draft_version_fallback(strategy: str, sphinx_config: Dict[str, Any]):
+def _get_draft_version_fallback(strategy: str, sphinx_config: SphinxConfig) -> str:
```
Comment (webknjaz) [inline src/sphinxcontrib/towncrier/__init__.py:70]:
```suggestion
def _get_draft_version_fallback(
        strategy: str,
        sphinx_config: SphinxConfig,
) -> str:
```

File: mypy.ini (hunk 2 above)
Comment (slsh1o) [reply mypy.ini]: Ok, I turned `disallow_untyped_calls` to True and will add other configs in another PR.

**Review by webknjaz — APPROVED** [review]
